akka-remote/src/main/scala/akka/remote/artery/tcp/SSLEngineProvider.scala

```diff
@@ -73,119 +62,66 @@ class ConfigSSLEngineProvider(protected val config: Config, protected val log: M
+    JksManagerProviders.loadKeystore(filename, password)
 
   /**
    * Subclass may override to customize `KeyManager`
    */
-  protected def keyManagers: Array[KeyManager] = {
-    val factory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
-    factory.init(loadKeystore(SSLKeyStore, SSLKeyStorePassword), SSLKeyPassword.toCharArray)
-    factory.getKeyManagers
-  }
+  @deprecated("Implement and setup a SslManagersProvider instead of overriding this method.", "2.6.6")
+  protected def keyManagers: Array[KeyManager] = keyStoreProviders.keyManagers
 
   /**
    * Subclass may override to customize `TrustManager`
    */
-  protected def trustManagers: Array[TrustManager] = {
-    val trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
-    trustManagerFactory.init(loadKeystore(SSLTrustStore, SSLTrustStorePassword))
-    trustManagerFactory.getTrustManagers
-  }
+  @deprecated("Implement and setup a SslManagersProvider instead of overriding this method.", "2.6.6")
+  protected def trustManagers: Array[TrustManager] = keyStoreProviders.trustManagers
 
-  def createSecureRandom(): SecureRandom =
-    SecureRandomFactory.createSecureRandom(SSLRandomNumberGenerator, log)
+  // TODO: remove
+  @deprecated("Use SecureRandomFactory directly.", "2.6.6")
+  def createSecureRandom(): SecureRandom = rng
 
   override def createServerSSLEngine(hostname: String, port: Int): SSLEngine =
-    createSSLEngine(akka.stream.Server, hostname, port)
+    sslFactory.createServerSSLEngine(hostname, port)
 
   override def createClientSSLEngine(hostname: String, port: Int): SSLEngine =
-    createSSLEngine(akka.stream.Client, hostname, port)
-
-  private def createSSLEngine(role: TLSRole, hostname: String, port: Int): SSLEngine = {
-    createSSLEngine(sslContext, role, hostname, port)
-  }
-
-  private def createSSLEngine(sslContext: SSLContext, role: TLSRole, hostname: String, port: Int): SSLEngine = {
-
-    val engine = sslContext.createSSLEngine(hostname, port)
-
-    if (HostnameVerification && role == akka.stream.Client) {
-      val sslParams = sslContext.getDefaultSSLParameters
-      sslParams.setEndpointIdentificationAlgorithm("HTTPS")
-      engine.setSSLParameters(sslParams)
-    }
-
-    engine.setUseClientMode(role == akka.stream.Client)
-    engine.setEnabledCipherSuites(SSLEnabledAlgorithms.toArray)
-    engine.setEnabledProtocols(Array(SSLProtocol))
-
-    if ((role != akka.stream.Client) && SSLRequireMutualAuthentication)
-      engine.setNeedClientAuth(true)
-
-    engine
-  }
+    sslFactory.createClientSSLEngine(hostname, port)
 
   override def verifyClientSession(hostname: String, session: SSLSession): Option[Throwable] =
-    None
+    NoopSessionVerifier.verifyClientSession(hostname, session)
 
   override def verifyServerSession(hostname: String, session: SSLSession): Option[Throwable] =
-    None
+    NoopSessionVerifier.verifyServerSession(hostname, session)
```

Comment:
Post handshake verifications are also extracted, and implementors of an `SSLEngineProvider` may reuse the existing ones.